src/sql/parts/notebook/notebookInput.ts

```diff
@@ -166,16 +218,19 @@ export class NotebookInput extends EditorInput {
 		return NotebookInput.ID;
 	}
 
-	public resolve(refresh?: boolean): TPromise<IEditorModel> {
-		return undefined;
+	getResource(): URI {
+		return this.resource;
 	}
 
-	public getName(): string {
-		if (!this._title) {
-			this._title = resources.basenameOrAuthority(this._model.notebookUri);
+	async resolve(): TPromise<NotebookEditorModel> {
+		if(this._model && this._model.isModelCreated()){
+			return TPromise.as(this._model);
+		}else{
+		const textEditorModelReference = await this.textModelService.createModelReference(this.resource);
+		const textEditorModel = await textEditorModelReference.object.load();
+		this._model = this.instantiationService.createInstance(NotebookEditorModel, this.resource, false, textEditorModel);
```

Comment:
Don't we need to plumb through the connectionProfileId and possibly the active provider which we did before? I think we need to pass them through / set them at this stage unless I"m missing something.